firmware/controllers/system/periodic_timer_controller.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * @file periodic_timer_controller.h
+ *
+ * @date Jul 8, 2019
+ * @author Andrey Belomutskiy, (c) 2012-2019
+ */
+
+#pragma once
+
+#include "global.h"
+
+class PeriodicTimerController;
+
+void runAndScheduleNext(PeriodicTimerController *controller);
+
+/**
+ * this is an intermediate implementation - we should probably move from using virtual_timer_t which works on interrupts
+ * into sharing one thread with potentially lower priority instead
+ */
+
+class PeriodicTimerController {
+
+public:
+#if !EFI_UNIT_TEST
+	virtual_timer_t timer;
+#endif /* EFI_UNIT_TEST */
+	int periodMs;
+
+	virtual void PeriodicTask() = 0;
```

Comment:
One option is to have the task's implementation return the period (change the signature here to `virtual uint32_t PeriodicTask() = 0`).  Since we can't actually do anything about period changes at any time other than immediately after the task runs, returning it helps convey that limitation (it also saves having to store the period somewhere).